lib/test/test_utf8mb4_extensions.rb

```diff
@@ -0,0 +1,84 @@
+require_relative 'test_helper'
+require_relative '../utf8mb4_extensions'
+
+describe CoreExtensions::String::Utf8mb4 do
+  let(:string) {"foobar"}
+
+  describe '.utf8mb4?' do
+    let(:utf8mb4?) {string.utf8mb4?}
+
+    context 'string has only latin characters' do
+      it 'returns false' do
+        _(utf8mb4?).must_equal false
+      end
+    end
+
+    context 'string has some three-byte characters' do
+      let(:string) {"汉语"}
```

Comment:
Nit:
Could we use a more generic way of defining a 4-byte charter? I'm not fluent in Chinese, maybe defining the characters by the unicode.
```suggestion
      let(:string) {"\u{6C49}\u{8BED}"} #汉语
```

Reply:
It might not be very obvious that the larger codepoint encodes to four bytes, but I've written them in as you suggest.

Reply:
Though, in general, codepoints `U+10000` and greater are 4 bytes... so this should be reasonable.

Reply:
Thanks!
I agree it's not too much different. I was looking for a very obvious way of differentiating which characters were 4-byte vs 3-byte and the Unicode was slightly more obvious.
I also wondered if it was a reasonable ask or not.